### PR TITLE
check_basepage_size:Add mapping hugepage check

### PR DESCRIPTION
**Pull request by zhenyzha**
1. Update basic page size to dynamic checking method
2. Add mapping hugepage check

ID:2185726
Signed-off-by: zhenyzha <zhenyzha@redhat.com>

### CODE REVIEW

**Comment by zhenyzha** [comment]
git am 3689.patch
Applying: check_basepage_size:Add mapping hugepage check

` (1/1) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.check_basepage_size.arm64-pci: PASS (39.48 s)`

@PaulYuuu Could you pls help review when you are free. thanks.

File: qemu/tests/check_basepage_size.py [hunk 1]
```diff
@@ -3,33 +3,53 @@
...
+    5) Check the basic page size mapping to the hugepage size on guest.
 
     :params test: QEMU test object.
     :params params: Dictionary with the test parameters.
     :params env: Dictionary with test environment.
     """
-    get_page_size = params.get('get_page_size')
+    get_basic_page = params.get('get_basic_page')
+    get_local_pglist = params.get('get_local_pglist')
+    basic_page_list = params.get('basic_page_list').split()
```
Comment (PaulYuuu) [inline qemu/tests/check_basepage_size.py:26]:
```suggestion
    basic_page_list = params.objects('basic_page_list')
```

File: qemu/tests/check_basepage_size.py [hunk 2]
```diff
@@ -3,33 +3,53 @@
...
-    host_page_size = process.system_output(get_page_size).decode()
-    if host_page_size != params.get("page_size_host"):
-        test.fail("Host page size is %s not as expected." % host_page_size)
+    host_basic_page = process.system_output(get_basic_page).decode()
+    if host_basic_page not in basic_page_list:
+        test.fail("Host basic page size is %s not as expected." % host_basic_page)
+
+    test.log.info("Check the basic page size mapping to the hugepage size on host.")
+    host_basic_page = normalize_data_size("%sB" % host_basic_page, "K")
+    hugepage_list = params.get('mapping_pgsize_%sk' % host_basic_page).split()
```
Comment (PaulYuuu) [inline qemu/tests/check_basepage_size.py:35]: Same above.

File: qemu/tests/check_basepage_size.py [hunk 3]
```diff
@@ -3,33 +3,53 @@
...
+    local_pglist = process.system_output(get_local_pglist).decode().split()
```
Comment (PaulYuuu) [inline qemu/tests/check_basepage_size.py:36]: `os.listdir`

File: qemu/tests/check_basepage_size.py [hunk 4]
```diff
@@ -3,33 +3,53 @@
...
+    guest_basic_page = session.cmd(get_basic_page).strip()
+    if guest_basic_page not in basic_page_list:
+        test.fail("Guest page size is %s not as expected." % guest_basic_page)
+
+    test.log.info("Check the basic page size mapping to the hugepage size on guest.")
+    guest_basic_page = normalize_data_size("%sB" % guest_basic_page, "K")
+    hugepage_list = params.get('mapping_pgsize_%sk' % guest_basic_page).split()
+    local_pglist = session.cmd(get_local_pglist).strip().split()
+    if sorted(local_pglist) != sorted(hugepage_list):
+        test.fail("Host huge page size is %s not as expected." % local_pglist)
```
Comment (PaulYuuu) [inline qemu/tests/check_basepage_size.py:55]: Guest?

File: qemu/tests/check_basepage_size.py (hunk 1 above)
Comment (zhenyzha) [reply qemu/tests/check_basepage_size.py]: thanks a lot, has been corrected.

File: qemu/tests/check_basepage_size.py (hunk 3 above)
Comment (zhenyzha) [reply qemu/tests/check_basepage_size.py]: thanks a lot, has been corrected.

File: qemu/tests/check_basepage_size.py (hunk 4 above)
Comment (zhenyzha) [reply qemu/tests/check_basepage_size.py]: thanks a lot, has been corrected.

**Comment by zhenyzha** [comment]
@PaulYuuu Hello Could you pls help review again when you are free? thanks.

git am 3651.patch
Applying: HugePageConfig:Add hugepages release operation

```
 (1/1) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.check_basepage_size.arm64-pci: STARTED
 (1/1) Host_RHEL.m9.u3.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.aarch64.io-github-autotest-qemu.check_basepage_size.arm64-pci: PASS (44.77 s)
```

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**